test/unit_test/character_sheet_unit_tests/FocusesUnitTests.java

```diff
@@ -0,0 +1,144 @@
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.CharacterSheet;
+import rpg_database.character_sheet.Fields;
+import rpg_database.character_sheet.Focuses;
+import rpg_database.character_sheet.FocusesSet;
+
+public class FocusesUnitTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void expectException_SetFocusesMalformedInput() {
+		expectExceptionWithMessage(InvalidParameterException.class,
+				"class java.lang.String value is not an instance of class rpg_database.character_sheet.Focuses");
+		final String malformedInput = "MALFORMED INPUT";
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.FOCUS, malformedInput);
+	}
+
+	// functional tests
+
+	@Test
+	public void testSetFocusWithHashSet() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		HashSet<Focuses> expectedFocuses = new HashSet<Focuses>();
+		expectedFocuses.add(Focuses.INTIMIDATION);
+		characterSheet.setData(Fields.FOCUS, new FocusesSet(Focuses.INTIMIDATION));
+		FocusesSet actualFocus = characterSheet.getData(Fields.FOCUS);
+		assertEquals(expectedFocuses, actualFocus);
+	}
+
+	@Test
+	public void testSetFocusWithFocusesSet() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		FocusesSet expectedFocuses = new FocusesSet(Focuses.ETIQUETTE);
+		characterSheet.setData(Fields.FOCUS, expectedFocuses);
+		FocusesSet actualFocus = characterSheet.getData(Fields.FOCUS);
+		assertEquals(expectedFocuses, actualFocus);
+	}
+
+	@Test
+	public void testSetMoreFocus() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		FocusesSet expectedFocuses = new FocusesSet(Focuses.ANIMAL_HANDLING, Focuses.ARCANE_LANCE, Focuses.BRAWLING);
+		characterSheet.setData(Fields.FOCUS, expectedFocuses);
+		FocusesSet actualFocuses = characterSheet.getData(Fields.FOCUS);
+		assertEquals(expectedFocuses, actualFocuses);
+	}
+
+	@Test
+	public void testSetSingleTimesFocus() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		FocusesSet tempFocuses = new FocusesSet(Focuses.ANIMAL_HANDLING);
+		characterSheet.setData(Fields.FOCUS, tempFocuses);
+		assertEquals(2, Focuses.ANIMAL_HANDLING.getFocusValue());
+	}
+
+	@Test
+	public void testSetMultipleTimesFocus() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		FocusesSet tempFocuses = new FocusesSet(Focuses.ANIMAL_HANDLING, Focuses.ANIMAL_HANDLING);
+		characterSheet.setData(Fields.FOCUS, tempFocuses);
+		assertEquals(3, Focuses.ANIMAL_HANDLING.getFocusValue());
+	}
+
+	private void expectExceptionWithMessage(Class<? extends Exception> exceptionClass, String message) {
+		thrown.expect(exceptionClass);
+		thrown.expectMessage(message);
+	}
+
+	// Majd a front-end get requestét egy ilyen logika kellene, hogy kisolgálja,

```

Comment:
Azért majd nézd őket át hogy a refaktor után melyik valid meg melyik nem, olyat ne rakj be feleslegesen amire már nincs szükség.